x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/openai/OpenAiUnifiedChatCompletionRequestEntity.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.startArray(MESSAGES_FIELD);
+        {
+            for (UnifiedCompletionRequest.Message message : unifiedRequest.messages()) {
+                builder.startObject();
+                {
+                    switch (message.content()) {
+                        case UnifiedCompletionRequest.ContentString contentString -> builder.field(CONTENT_FIELD, contentString.content());
+                        case UnifiedCompletionRequest.ContentObjects contentObjects -> {
+                            builder.startArray(CONTENT_FIELD);
+                            for (UnifiedCompletionRequest.ContentObject contentObject : contentObjects.contentObjects()) {
+                                builder.startObject();
+                                builder.field(TEXT_FIELD, contentObject.text());
+                                builder.field(TYPE_FIELD, contentObject.type());
+                                builder.endObject();
+                            }
+                            builder.endArray();
+                        }
+                    }
+
+                    builder.field(ROLE_FIELD, message.role());
+                    if (message.name() != null) {
+                        builder.field(NAME_FIELD, message.name());
+                    }
+                    if (message.toolCallId() != null) {
+                        builder.field(TOOL_CALL_ID_FIELD, message.toolCallId());
+                    }
+                    if (message.toolCalls() != null) {
+                        builder.startArray(TOOL_CALLS_FIELD);
+                        for (UnifiedCompletionRequest.ToolCall toolCall : message.toolCalls()) {
+                            builder.startObject();
+                            {
+                                builder.field(ID_FIELD, toolCall.id());
+                                builder.startObject(FUNCTION_FIELD);
+                                {
+                                    builder.field(ARGUMENTS_FIELD, toolCall.function().arguments());
+                                    builder.field(NAME_FIELD, toolCall.function().name());
+                                }
+                                builder.endObject();
+                                builder.field(TYPE_FIELD, toolCall.type());
+                            }
+                            builder.endObject();
+                        }
+                        builder.endArray();
+                    }
+                }
+                builder.endObject();
+            }
+        }
+        builder.endArray();
+
+        builder.field(MODEL_FIELD, model.getServiceSettings().modelId());
+        if (unifiedRequest.maxCompletionTokens() != null) {
+            builder.field(MAX_COMPLETION_TOKENS_FIELD, unifiedRequest.maxCompletionTokens());
```

Comment:
I just realized that the OpenAiChatCompletionServiceSettings has a similar field that isn't used (even previous to this PR). I wonder if we should sync those fields up like we do the `modelId` 🤔 . I think we've typically used the max tokens to do the truncation for text embedding so I don't think we really used it for completions.

@maxhniebergall what do you think?

Reply:
hmm, yea it sounds like we should use the value from the service settings if its available